goiardi/plan.sh

```diff
@@ -0,0 +1,48 @@
+pkg_name=goiardi
+pkg_origin=core
+pkg_version=0.11.10
+pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
+pkg_license=("Apache-2.0")
+pkg_description="A Chef server written in Go"
+pkg_upstream_url="http://goiardi.gl"
+# pkg_scaffolding="some/scaffolding"
+pkg_source="https://github.com/ctdk/goiardi/archive/v${pkg_version}.tar.gz"
+# pkg_filename="${pkg_name}-${pkg_version}.tar.gz"
+pkg_shasum=74bb265eaef7d9d470a8ccce7bef651aaff41e314cd177a11ad6493ff0179345
+pkg_deps=(
+  core/glibc
+)
+pkg_build_deps=(
+  core/go
+  core/git
+)
+pkg_bin_dirs=(bin)
+# pkg_svc_run="haproxy -f $pkg_svc_config_path/haproxy.conf"
+# pkg_exports=(
+#   [host]=srv.address
+#   [port]=srv.port
+#   [ssl-port]=srv.ssl.port
+# )
+# pkg_exposes=(port ssl-port)
+# pkg_binds=(
+#   [database]="port host"
+# )
+# pkg_binds_optional=(
+#   [storage]="port host"
+# )
+# pkg_interpreters=(bin/bash)
+# pkg_svc_user="hab"
+# pkg_svc_group="$pkg_svc_user"
+
+do_build() {
+  local build_dir="${HAB_CACHE_SRC_PATH}/goiardi-${pkg_version}"
+  mkdir -p "${build_dir}"
+  pushd "${build_dir}" > /dev/null
+  go mod init
+  go build
```

Comment:
I noticed that the upstream repository has a vendor directory:

https://github.com/ctdk/goiardi/tree/master/vendor

I wonder if we might want to use `-mod=vendor` here to make sure we are using the vendored copies of the dependencies.

Reply:
The vendored copies may  or may not be up to date. I don't know what the "best" approach here is. My validation is just that the resulting binaries run as expected, and they do so without using the vendored deps.

Reply:
The project also doesn't have a `go.mod` so, what is the reasoning to create one here?